RHIDP-5901: readOnlyRootFilesystem option (#938)
redhat-developer/red-hat-developers-documentation-rhdh: assemblies/assembly-configuring-readonlyrootfilesystem.adoc

:_mod-docs-content-type: ASSEMBLY
:context: readonlyrootfilesystem
[id="{context}"]
= Configuring readOnlyRootFilesystem in {product}

> The {product} deployment consists of two containers, an `initContainer` that installs the Dynamic Plugins, and a backend container which runs the application. The `initContainer` has the `readOnlyRootFilesystem` option enabled by default. To enable this option, you must have `platform administrator` permissions so that you can modify the configuration and redeploy. You can manually configure the `readOnlyRootFilesystem` option on the backend container by using the following methods.

Reviewer: ```suggestion
The {product} deployment consists of two containers, an `initContainer` that installs the Dynamic Plugins, and a backend container which runs the application. The `initContainer` has the `readOnlyRootFilesystem` option enabled by default. To enable this option on the backend container, you must have `platform administrator` permissions so that you can modify the configuration and redeploy. You can manually configure the `readOnlyRootFilesystem` option on the backend container by using the following methods.
```

Otherwise, it seems like the option is not enabled on the init container, IMO.
Also, I am not sure `platform administrator` permissions are needed. I only need permission to deploy resources via Helm or permission to create/update a Custom Resource for Operator-backed deployments.

> The {product} deployment consists of two containers, an `initContainer` that installs the Dynamic Plugins, and a backend container which runs the application. The `initContainer` has the `readOnlyRootFilesystem` option enabled by default. To enable this option on the backend container, you must either have permission to deploy resource through Helm or to create or update a CR for Operator-backed deployments. You can manually configure the `readOnlyRootFilesystem` option on the backend container by using the following methods:

Reviewer 2: ```suggestion
The {product} deployment consists of two containers: an `initContainer` that installs the Dynamic Plugins and a backend container that runs the application. The `initContainer` has the `readOnlyRootFilesystem` option enabled by default. To enable this option on the backend container, you must either have permission to deploy resources through Helm or to create or update a CR for Operator-backed deployments. You can manually configure the `readOnlyRootFilesystem` option on the backend container by using the following methods:

* The {product} Operator
* The {product} Helm chart
```

> The {product} deployment consists of two containers: an `initContainer` that installs the Dynamic Plugins, and a backend container that runs the application. The `initContainer` has the `readOnlyRootFilesystem` option enabled by default. To enable this option on the backend container, you must either have permission to deploy resources through Helm or to create or update a CR for Operator-backed deployments. You can manually configure the `readOnlyRootFilesystem` option on the backend container by using the following methods:

Reviewer 2: ```suggestion
The {product} deployment consists of two containers: an `initContainer` that installs the Dynamic Plugins, and a backend container that runs the application. The `initContainer` has the `readOnlyRootFilesystem` option enabled by default. To enable this option on the backend container, you must either have permission to deploy resources through Helm or to create or update a CR for Operator-backed deployments. You can manually configure the `readOnlyRootFilesystem` option on the backend container by using the following methods:

* The {product} Operator
* The {product} Helm chart
```

I see that we are trying to introduce what the methods are by dropping the Operator and Helm chart modules after the colon, but I do recommend adding the bullets to close out the module with a complete statement. Also, without this little list, the user would have to get through the entire Operator module before they learned that the Helm chart is the other method.
